gapir/cc/context.cpp

```diff
@@ -231,101 +231,77 @@ void Context::registerCallbacks(Interpreter* interpreter) {
       Gles::INDEX, Builtins::ReplayFrameDelimiter,
       [this](uint32_t label, Stack* stack, bool) {
         uint32_t id = stack->pop<uint32_t>();
-        if (stack->isValid()) {
-          GAPID_INFO("[%u]replayFrameDelimiter(%u)", label, id);
-          auto found = mGlesRenderers.find(id);
-          if (found == mGlesRenderers.end()) {
-            GAPID_ERROR("replayFrameDelimiter has no renderer at ID: %u", id);
-            return false;
-          }
-          found->second->frameDelimiter();
-          return true;
-        } else {
-          GAPID_WARNING(
-              "[%u]Error during calling function replayFrameDelimiter", label);
+        GAPID_INFO("[%u]replayFrameDelimiter(%u)", label, id);
+        auto found = mGlesRenderers.find(id);
+        if (found == mGlesRenderers.end()) {
+          GAPID_ERROR("replayFrameDelimiter has no renderer at ID: %u", id);
           return false;
         }
+        found->second->frameDelimiter();
+        return true;
       });
 
   interpreter->registerBuiltin(
       Gles::INDEX, Builtins::ReplayCreateRenderer,
       [this](uint32_t label, Stack* stack, bool) {
         uint32_t id = stack->pop<uint32_t>();
-        if (stack->isValid()) {
-          GAPID_INFO("[%u]replayCreateRenderer(%u)", label, id);
-          auto existing = mGlesRenderers.find(id);
-          if (existing != mGlesRenderers.end()) {
-            delete existing->second;
-          }
-          // Share objects with the root GLES context.
-          // This will essentially make all objects shared between all contexts.
-          // It is ok since correct replay will only reference what it is
-          // supposed to.
+        GAPID_INFO("[%u]replayCreateRenderer(%u)", label, id);
```

Comment:
So what happens now if the stack is not valid?

Reply:
Undefined. Almost certainly a crash. The checking was slowing down replay in a way that was impeding the functional usefulness of the code as a profiler. If you are interested I could leave the checking in for debug builds only. As long as we make sure to generate valid VM bytecode the stack should never become invalid. Processors and operating systems don't typically check their stacks for validity for the same reason. Performance trumps security in a profiling tool like the one we're making.

Reply:
We should at least keep these in in a debug build. It is too easy to make a mistake when generating the instructions, that having the check is valuable (this is why it exists in the first place).

Reply:
We should definitely leave these in at least in debug. It helps a lot when we run into the "Generating invalid VM bytecode" case, which can certainly happen if we are in an invalid case.